src/jobs/jobs.controller.ts

```diff
@@ -369,21 +356,21 @@ export class JobsController {
           HttpStatus.BAD_REQUEST,
         );
       }
-    } else if (jobCreateDto.jobParams.auth == JobsAuth.DatasetAccess) {
+    } else if (jobParams.auth == JobsAuth.DatasetAccess) {
       // verify that all the pids listed in the property indicated are accessible by the user
-      const datasetIds = this.checkDatasetIds(jobCreateDto);
+      const datasetIds = this.checkDatasetIds(jobParams);
       const numberOfDatasets = await this.datasetsService.count({
         where: {
-          pid: { in: datasetIds },
+          pid: { $in: datasetIds },
           $or: [
-            { ownerGroup: { in: user.currentGroups } },
-            { accessGroups: { in: user.currentGroups } },
-            { isPublished: true }  // TODO
+            { ownerGroup: { $in: user.currentGroups } },
+            { accessGroups: { $in: user.currentGroups } },
+            { isPublished: true }  // TBD
           ]     
         },
       });
 
-      const datasetsNoAccess = datasetIds.length - numberOfDatasets.count;
+      const datasetsNoAccess = datasetIds.length - numberOfDatasets.count; // TBD
```

Comment:
The current implementation to check access to datasets was copied from the previous version.

Why was it written like this? Is it sufficient or should it change?

Reply:
In one of our discussions, we agreed that the job creation will succeed if the user has access to all the requested datasets.
I would keep it this way for now